packages/cli/src/UserManagement/routes/passwordReset.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable import/no-cycle */
+
+import express = require('express');
+import { v4 as uuidv4 } from 'uuid';
+import { URL } from 'url';
+import { genSaltSync, hashSync } from 'bcryptjs';
+
+import { Db, ResponseHelper } from '../..';
+import { N8nApp } from '../Interfaces';
+import { isValidEmail, validatePassword } from '../UserManagementHelper';
+import * as UserManagementMailer from '../email';
+import type { PasswordResetRequest } from '../Interfaces';
+import { issueJWT } from '../auth/jwt';
+
+export function addPasswordResetNamespace(this: N8nApp): void {
+	/**
+	 * Send a password reset email.
+	 */
+	this.app.post(
+		`/${this.restEndpoint}/forgot-password`,
+		ResponseHelper.send(async (req: PasswordResetRequest.Email) => {
+			const { email } = req.body;
+
+			if (!email) {
+				throw new ResponseHelper.ResponseError('Email is mandatory', undefined, 400);
+			}
+
+			if (!isValidEmail(email)) {
+				throw new ResponseHelper.ResponseError('Invalid email address', undefined, 400);
+			}
+
+			if (!req.headers.host) {
+				throw new ResponseHelper.ResponseError('No host found', undefined, 400);
+			}
+
+			const user = await Db.collections.User!.findOne({ email });
+
+			if (!user) {
+				throw new ResponseHelper.ResponseError('', undefined, 404);
+			}
+
+			user.resetPasswordToken = uuidv4();
+
+			const { id, firstName, lastName, resetPasswordToken } = user;
+
+			await Db.collections.User!.update(id, { resetPasswordToken });
+
+			const baseUrl = `${req.protocol}://${req.headers.host}`;
+			const url = new URL(`/${this.restEndpoint}/resolve-password-token`, baseUrl);
```

Comment:
I think we need to get the path from mutasem then

Reply:
Got it, let me know what you decide :)